torch/csrc/distributed/c10d/ProcessGroupGloo.cpp

```diff
@@ -465,9 +469,29 @@ ProcessGroupGloo::AsyncWork::AsyncWork(
     std::vector<std::vector<at::Tensor>> outputTensors,
     const char* profilingTitle,
     const c10::optional<std::vector<at::Tensor>>& inputTensors)
-    : ProcessGroup::Work(-1, OpType::UNKNOWN, profilingTitle, inputTensors),
+    : ProcessGroup::Work(-1, OpType::UNKNOWN, nullptr, inputTensors),
```

Comment:
Specify nullptr to profilingTitle of Parent Class is to disable the profiling in Parent Class. Because we will add new record function profiling in this child class to replace it.

Reply:
+1, can you mention that for general ProcessGroup profiling is kicked off upon construction of `Work` object, but it is delayed for gloo and why? Thanks!

Reply:
Added. thx!